pkg/protocompat/time.go

```diff
@@ -73,5 +73,5 @@ var (
 
 // IsZeroTimestamp returns whether a Timestamp pointer is either nil, or pointing to the zero of the type.
 func IsZeroTimestamp(ts *gogoTimestamp.Timestamp) bool {
-	return ts == nil || ts == &gogoTimestamp.Timestamp{} || ts == zeroProtoTimestampFromTime
+	return ts == nil || Equal(ts, &gogoTimestamp.Timestamp{}) || Equal(ts, zeroProtoTimestampFromTime)
```

Comment:
nit: maybe use `GetProtoTimestampZero` instead of `&gogoTimestamp.Timestamp{}` so it's one fewer thing to change when moving away from the gogo library. Plus it'll be consistent with what users of this package will use